extensions/applicationinsights-clickanalytics-js/src/common/Utils.ts

```diff
@@ -265,50 +265,50 @@ export function validateContentNamePrefix ( config: IClickAnalyticsConfiguration
-            referrerUri: hasDocument ? document.referrer : "",
-            requestUri: "",
-            pageName: "",
-            pageType: ""
-        },
-        dataTags: {
-            useDefaultContentNameOrId: false,
-            aiBlobAttributeTag: DEFAULT_AI_BLOB_ATTRIBUTE_TAG,
-            customDataPrefix: DEFAULT_DATA_PREFIX,
-            captureAllMetaDataContent: false,
-            dntDataTag: DEFAULT_DONOT_TRACK_TAG
-        },
-        behaviorValidator: (key:string) => key || "",
-        defaultRightClickBhvr: "",
-        dropInvalidEvents : false
-    };
-
-    let attributesThatAreObjectsInConfig: any[] = [];
-    for (const attribute in defaultConfig) {
-        if (typeof defaultConfig[attribute] === "object") {
-            attributesThatAreObjectsInConfig.push(attribute);
-        }
-    }
-
-    if (overrideConfig) {
-        // delete attributes that should be object and
-        // delete properties that are null, undefined, ''
-        removeNonObjectsAndInvalidElements(overrideConfig, attributesThatAreObjectsInConfig);
-        if(isValueAssigned(overrideConfig.dataTags)) {
-            overrideConfig.dataTags.customDataPrefix = validateContentNamePrefix(overrideConfig, DEFAULT_DATA_PREFIX) ? overrideConfig.dataTags.customDataPrefix : DEFAULT_DATA_PREFIX;
-        }
-        return objExtend(true, defaultConfig, overrideConfig);
-    }
-}
+// export function mergeConfig(overrideConfig: IClickAnalyticsConfiguration): IClickAnalyticsConfiguration {
+//     let defaultConfig: IClickAnalyticsConfiguration = {
+//         // General library settings
+//         autoCapture: true,
+//         callback: {
+//             pageActionPageTags: null
+//         },
+//         pageTags: {},
+//         // overrideValues to use instead of collecting automatically
+//         coreData: {
+//             referrerUri: hasDocument ? document.referrer : "",
+//             requestUri: "",
+//             pageName: "",
+//             pageType: ""
+//         },
+//         dataTags: {
+//             useDefaultContentNameOrId: false,
+//             aiBlobAttributeTag: DEFAULT_AI_BLOB_ATTRIBUTE_TAG,
+//             customDataPrefix: DEFAULT_DATA_PREFIX,
+//             captureAllMetaDataContent: false,
+//             dntDataTag: DEFAULT_DONOT_TRACK_TAG
+//         },
+//         behaviorValidator: (key:string) => key || "",
+//         defaultRightClickBhvr: "",
+//         dropInvalidEvents : false
+//     };
+
+//     let attributesThatAreObjectsInConfig: any[] = [];
+//     for (const attribute in defaultConfig) {
+//         if (typeof defaultConfig[attribute] === "object") {
+//             attributesThatAreObjectsInConfig.push(attribute);
+//         }
+//     }
+
+//     if (overrideConfig) {
+//         // delete attributes that should be object and
+//         // delete properties that are null, undefined, ''
+//         removeNonObjectsAndInvalidElements(overrideConfig, attributesThatAreObjectsInConfig);
```

Comment:
The new version of this is that each "field" can define an `isVal` which is a callback function that is past the user supplied "value" and if `false` the default will be assigned otherwise the default value will attempt to be merged with this value.

So we should probably use this for these cases, and because of the dynamic nature of the config now we don't want to "remove" the invalid values.